core/trino-main/src/main/java/io/trino/metadata/GlobalFunctionCatalog.java

```diff
@@ -180,13 +177,20 @@ public ScalarFunctionImplementation getScalarFunctionImplementation(
     @Override
     public TableFunctionProcessorProvider getTableFunctionProcessorProvider(ConnectorTableFunctionHandle functionHandle)
     {
-        if (functionHandle instanceof ExcludeColumnsFunctionHandle) {
-            return getExcludeColumnsFunctionProcessorProvider();
-        }
-        if (functionHandle instanceof SequenceFunctionHandle) {
-            return getSequenceFunctionProcessorProvider();
-        }
+        // TODO somehow to inject tableFunctions here
+        // TODO TableFunctionHandle doesn't contains name anymore https://github.com/trinodb/trino/pull/17117
+        // need somehow to map ConnectorTableFunctionHandle to TableFunctionHandle/ConnectorTableFunction
+        ConnectorTableFunctionHandle connectorTableFunctionHandle = null;
 
+        if (connectorTableFunctionHandle instanceof AbstractConnectorTableFunctionExtended abstractConnectorTableFunctionExtended) {
```

Comment:
remove `AbstractConnectorTableFunctionExtended` interface

let GFC get a list of TableFunctionProcessorProvider factories. the factories will be called in sequence and only one can return a TableFunctionProcessorProvider

perhaps
```
return  tableFunctionProcessorFactories.stream()
    .map(factory -> getTableFunctionProcessorProvider(factory))
   .flaMap(Optional::stream)
    .collect(toOptional())
   .orElse(null);
```

Reply:
@findepi big thanks, refactored